Source/NSURLResponse.m

```diff
@@ -138,6 +138,18 @@ - (void) _setHeaders: (id)headers
 	      n = [NSString stringWithFormat: @"%@, %@", o, n];
 	    }
 	  [self _setValue: v forHTTPHeaderField: n];
+
+          // If key == "Set-Cookie"/"Set-Cookie2" there could be multiple ones...
```

Comment:
Here the indentation is wrong.
As for the logic here I am not sure what it does, or rather what it should do. First containsString: should most likely be hasPrefix:. The existing code above also seems to try to merge values for the same key. But there we seem to mix values and keys. Maybe somebody needs to write a few test cases for this code to get the desired behaviour better defined?